src/bp/core/services/bot-service.ts

```diff
@@ -138,6 +141,44 @@ export class BotService {
     }
   }
 
+  async exportBot(botId: string): Promise<Buffer> {
+    return this.ghostService.forBot(botId).exportToArchiveBuffer()
+  }
+
+  async importBot(botId: string, archive: Buffer, allowOverwrite?: boolean): Promise<void> {
+    const alreadyExists = (await this.getBotsIds()).includes(botId)
+    if (alreadyExists) {
+      if (!allowOverwrite) {
+        return this.logger.error(`Cannot import the bot ${botId}, it already exists, and overwrite is not allowed`)
+      } else {
+        this.logger.warn(`The bot ${botId} already exists, files in the archive will overwrite existing ones`)
+      }
+    }
+    const tmpDir = tmp.dirSync({ unsafeCleanup: true })
+    const tmpFolder = tmpDir.name
+
+    try {
+      await extractArchive(archive, tmpFolder)
+      const api = await createForGlobalHooks()
+
+      const hookResult = {
+        allowImport: true
+      }
+
+      await this.hookService.executeHook(new Hooks.BeforeBotImport(api, botId, tmpFolder, hookResult))
+
+      if (hookResult.allowImport) {
+        await this.ghostService.forBot(botId).importFromDirectory(tmpDir.name)
```

Comment:
At this point the bot is not created yet, how does scoped ghost behaves with an bot Id that "does't exist". I guess it makes no difference..

Reply:
scoped ghost only sets the folder where methods will run, so it doesn't care if it exists or not